modules/core/src/test/kotlin/datamaintain/core/report/ReportTest.kt

```diff
@@ -81,6 +103,74 @@ internal class ReportTest {
                 )
             }
         }
+
+        @Test
+        fun `should print only summary if neither verbose nor porcelain are set`() {
+            // Given
+            val report = reportFixture()
+
+            // When
+            report.print(verbose = false, porcelain = false)
+
+            // Then
+            expectThat(testAppender.events).map { it.message }.containsExactly(listOf(
+                    "Summary => ",
+                    "- 1 files scanned",
+                    "- 1 files filtered",
+                    "- 1 files pruned",
+                    "- 1 check rules validated",
+                    "- 2 files executed",
+                    " -> script1",
+                    " -> script2"
+            ))
+        }
+
+
+        @Test
+        fun `should print details if verbose is set`() {
+            // Given
+            val report = reportFixture()
+
+            // When
+            report.print(verbose = true, porcelain = false)
+
+            // Then
+            expectThat(testAppender.events).map { it.message }.containsExactly(listOf(
+                    "Summary => ",
+                    "- 1 files scanned",
+                    " -> Scanned Script 1",
+                    "- 1 files filtered",
+                    " -> Filtered Script 1",
+                    "- 1 files pruned",
+                    " -> Pruned Script 1",
+                    "- 1 check rules validated",
+                    " -> AlwaysSucceed",
+                    "- 2 files executed",
+                    " -> script1",
+                    " -> script2"
+            ))
+        }
+
+        @Test
+        fun `should print only up to max step`() {
+            // Given
+            val report = reportFixture()
+
+            // When
+            report.print(verbose = true, porcelain = false, maxStepToShow = Step.PRUNE)
+
+            // Then
+            expectThat(testAppender.events).map { it.message }.containsExactly(listOf(
+                    "Summary => ",
+                    "- 1 files scanned",
+                    " -> Scanned Script 1",
+                    "- 1 files filtered",
+                    " -> Filtered Script 1",
+                    "- 1 files pruned",
+                    " -> Pruned Script 1"
+            ))
+        }
```

Comment:
This test is great, I think there should be one for each step to ensure the correct steps are always printed depending on the max step reached.

Reply:
Ok, I'll add the missing ones, as well.